app/models/energy/ScheduleBase.py

```diff
@@ -0,0 +1,100 @@
+"""Schedule Base Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+
+
+class ScheduleContinuous(BaseModel):
+    """This Numeric Type allows all numbers, including fractional amounts, within the range to
+  be valid."""
+
+    type: Enum('ScheduleContinuous', {'type': 'ScheduleContinuous'})
+
+    schedule_continuous: float = None
+
+
+class ScheduleDiscrete(BaseModel):
+    """This Numeric Type allows all only integers within the range to be valid."""
+
+    type: Enum('ScheduleDiscrete', {'type': 'ScheduleDiscrete'})
+
+    schedule_discrete: int = None
+
+class ScheduleNumericType (BaseModel):
+    """Designates how the range values are validated."""
+
+    numeric_type:  Union[ScheduleContinuous, ScheduleDiscrete]
+
+
+class ScheduleUnitType (str, Enum):
+    dimensionless = 'Dimensionless'
+    temperature = 'Temperature'
+    delta_temperature = 'DeltaTemperature'
+    precipitation_rate = 'PrecipitationRate'
+    angle = 'Angle'
+    convection_coefficient = 'ConvectionCoefficient'
+    activity_level = 'ActivityLevel'
+    velocity = 'Velocity'
+    capacity = 'Capacity'
+    power = 'Power'
+    availability = 'Availability'
+    percent = 'Percent'
+    control = 'Control'
+    mode = 'Mode'
+
+
+class Date(BaseModel):
+    """Date."""
+
+    month: int = Schema(
+        1,
+        description='A value for month between `1`-`12`. Default is `1`.',
+        ge=1,
+        le=12
+    )
+
+    day: int = Schema(
+        1,
+        description='A value for day between `1`-`31`. Default is `1`.',
+        ge=1,
+        le=31
+    )
+
+
+class Time(BaseModel):
+    """Time."""
+
+    hour: int = Schema(
+        0,
+        description='A value for hour between `0`-`24`. Default is `0`.',
+        ge=0,
+        le=24,
+    )
+
+    minute: int = Schema(
+        0,
+        description='A value for month between `0`-`60`. Default is `0`.',
+        ge=0,
+        le=60,
+    )
+
+
+class DateTime(BaseModel):
```

Comment:
Why not just use the definition in `app/models/common/datetime.py`?

Reply:
I was using the `time` class by itself in https://github.com/ladybug-tools-in2/energy-model-schema/blob/fe50743c68915d1fe78579c2437f95e1c4626241/app/models/energy/ScheduleRuleset.py#L38

Reply:
I support how @tanushree04  is using it now but I think we can get rid of the DateTime object and have separate inputs of Dates and Times in the ScheduleRuleset.

Reply:
Update: I realized that, even if we have the date and time objects separate, we can still benefit from using Python's builtin `datetime` module to check the inputs here.  This is because there are separate objects for `date` and `time` within the within the `datetime` module.  So, all that you should have to do, @tanushree04 , to make sure that the user has input a valid date is this:
```
input_date = datetime.date(year=2016, day=day, month=month)
```
and this to test if the time is valid:
```
input_time = fatetime.time(hour=hour, minute=minute)
```
The one exception that we will have to catch is for time in the event that someone puts in 24:00 since it seems that we need to support this as a valid time to deal with EnergyPlus's quirks.